lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -2737,6 +2870,248 @@ void ChargePointImpl::handleGetLocalListVersionRequest(ocpp::Call<GetLocalListVe
     this->send<GetLocalListVersionResponse>(call_result);
 }
 
+DataTransferResponse ChargePointImpl::handle_set_user_price(const std::optional<std::string>& msg) {
+    std::optional<std::string> id_token = std::nullopt;
+    DataTransferResponse response;
+    response.status = DataTransferStatus::Rejected;
+    if (!msg.has_value()) {
+        EVLOG_error << "Data transfer for set user price does not contain any data.";
+        return response;
+    }
+
+    if (set_display_message_callback == nullptr) {
+        EVLOG_error << "Received data transfer for set user price, but now callback is registered.";
+        return response;
+    }
+
+    json data;
+    try {
+        data = json(msg.value());
```

Comment:
The 'SetUserPrice' translates into display messages here. Because it is in the end a message that should be displayed for a specific transaction. So do you suggest a new type instead of DisplayMessage or a conversion function with the json as input and a vector of display messages as output?

Reply:
Do you mean the EVerest DisplayMessage type? This should not be of interest in libocpp. My idea was to define a `struct` thats initialized by parsing the json instead of operating on the raw json